front/components/assistant/AssistantList.tsx

```diff
@@ -0,0 +1,65 @@
+import { AssistantPreview } from "@dust-tt/sparkle";
+import type {
+  ConversationType,
+  LightAgentConfigurationType,
+} from "@dust-tt/types";
+import { useRouter } from "next/router";
+import React from "react";
+
+interface AssistantListProps {
+  agents: LightAgentConfigurationType[];
+  handleAssistantClick: (
+    agent: LightAgentConfigurationType,
+    conversation?: ConversationType
+  ) => void;
+}
+
+const AssistantList: React.FC<AssistantListProps> = ({
```

Comment:
I've only abstracted it for readability here. 
Re: using it elsewhere, we could, but the gallery relies on GalleryAssistantPreviewContainer that has fairly different responsibilities (e.g. testing an assistant, sharing actions...). I'd be in favor to keep them separate for maintainability and to avoid excessive prop drilling.